sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3198,7 +3198,11 @@ class AstBuilder extends DataTypeAstBuilder with SQLConfHelper with Logging {
         commentSpec = Some(spec)
       }
     }
-
+    val dataType = typedVisit[DataType](ctx.dataType)
+    if (!SQLConf.get.collationEnabled &&
+      dataType.isInstanceOf[StringType] && !dataType.asInstanceOf[StringType].isDefaultCollation) {
```

Comment:
we should use `dataType.exists`, to check nested field data type as well.

Reply:
I only managed to find only existsRecursively, is that good?